### ISSUE TITLE
Implementing in Swift 3.0

### ISSUE THREAD

**Issue by grahammccarthy** [seaburg/IGIdenticon#7]
Hello, I'm trying to implement this library in Swift 3.0.

When setting up the generator, I come across the following problem:
```swift
let imageGenerator = IGImageGenerator(imageProducer: IGGitHubIdenticon(), hashFunction: IGJenkinsHashFromData(data: Data!))
````

I'm not sure how to implement the IGJenkinsHashFromData method into swift, can you please assist?

Below is the Objective-C implementation.

``` swift
#import "IGHashFunctions.h"

uint32_t IGJenkinsHashFromData(NSData *data)
{
    uint32_t hash = 0;
    const unsigned char *bytes = data.bytes;

    for(NSUInteger i = 0; i < [data length]; ++i) {
        hash += bytes[i];
        hash += (hash << 10);
        hash ^= (hash >> 6);
    }
    hash += (hash << 3);
    hash ^= (hash >> 11);
    hash += (hash << 15);

    return hash;
}
```

**Comment by seaburg** [comment]
Hello!
Try it:
```
func jenkinsHash(from data: Data) -> UInt32 {
    return data.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) -> UInt32 in
        var hash: UInt32 = 0
        for i in 0..<data.count {
            hash = hash &+ UInt32(bytes[i])
            hash = hash &+ (hash << 10);
            hash ^= (hash >> 6);
        }
        hash = hash &+ (hash << 3);
        hash ^= (hash >> 11);
        hash = hash &+ (hash << 15);

        return hash
    }
}
```

**Comment by grahammccarthy** [comment]
Thanks for writing the method.
It still doesn't work for me.

```swift
func configureCell(forChannel channel:Channel) {
        self.nameLabel.text = channel.name
        
        var data : Data = Data()
        let imageGenerator = IGImageGenerator(imageProducer: IGGitHubIdenticon(), hashFunction: jenkinsHash(from: data))
        self.avatarImage.image = imageGenerator?.image(from: channel.name, size: self.avatarImage.bounds.size)
    }
    
    func jenkinsHash(from data: Data?) -> UInt32 {
        return data!.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) -> UInt32 in
            var hash: UInt32 = 0
            for i in 0..<data!.count {
                hash = hash &+ UInt32(bytes[i])
                hash = hash &+ (hash << 10);
                hash ^= (hash >> 6);
            }
            hash = hash &+ (hash << 3);
            hash ^= (hash >> 11);
            hash = hash &+ (hash << 15);
            
            return hash
        }
    }
```

I get this error still:
```
Cannot convert value of type 'UInt32' to expected argument type '(@convention(c) (Data?) -> UInt32)!'
```

**Comment by seaburg** [comment]
Sorry. Fixed in 0.3.1

**Comment by grahammccarthy** [comment]
Yep! it works now.

Thanks @seaburg !
